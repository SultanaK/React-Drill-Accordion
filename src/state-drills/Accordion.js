import React from 'react';


class Accordions extends React.Component{
    static defaultProps = {
        sectionss: []
    };
    state = {
        activeSectionIndex: 0,
    };
    handleSetActiveSection = (sectionIndex) => {
        this.setState({
            activeSectionIndex: sectionIndex
        }); 
    };
    renderItem(section, idx, activeSectionIndex) {
            return (<li className = 'Accordion__item'
                key = {idx}>
                    <button type = 'button'
                    onClick = {() => this.handleSetActiveSection(idx)} >
                    { section.title} </button> 
                    {(activeSectionIndex === idx) && <p> {section.content} </p>} 
                </li>
                )
            }

    render() {
       const { activeSectionIndex} = this.state
       const {sections} = this.props
       return ( 
           <ul className ='Accordion'>{
               sections.map((section, idx) =>
                   this.renderItem(section, idx, activeSectionIndex)
               )
           } </ul>
       )
       }

    
}
export default Accordions;