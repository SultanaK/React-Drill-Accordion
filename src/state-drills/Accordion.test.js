import React from 'react';
//import ReactDOM from 'react-dom';
import Accordion from './Accordion';
//import renderer from 'react-test-renderer';
import {
    shallow
} from "enzyme";
import toJson from "enzyme-to-json";
import Accordions from './Accordion';



describe(`Tabs Component`, () => {
    const sections = [{
            title: 'Section 1',
            content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
            title: 'Section 2',
            content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
            title: 'Section 3',
            content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
    ];
    
    it("renders empty given no tabs", () => {
        const wrapper = shallow(<Accordion /> );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders the first tab by default", () => {
        const wrapper = shallow(<Accordion sections = {sections} />);
            expect(toJson(wrapper)).toMatchSnapshot();
                });

    it("closes the first tab and opens any clicked tab", () => {
         const wrapper = shallow(<Accordion sections = {sections} />)
             wrapper.find('button').at(1).simulate('click') 
             expect(toJson(wrapper)).toMatchSnapshot()
             });
    it("only open one Accordion at a time", () => {
            const wrapper = shallow(<Accordion sections = {sections} />)
                    wrapper.find('button').at(1).simulate('click')
                    wrapper.find('button').at(2).simulate('click')
                    expect(toJson(wrapper)).toMatchSnapshot()
                });

});