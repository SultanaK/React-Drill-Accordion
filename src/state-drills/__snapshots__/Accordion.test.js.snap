// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component closes the first tab and opens any clicked tab 1`] = `
<ul
  className="Accordion"
>
  <li
    className="Accordion__item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
       
    </button>
  </li>
  <li
    className="Accordion__item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
       
    </button>
    <p>
       
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
       
    </p>
  </li>
  <li
    className="Accordion__item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
       
    </button>
  </li>
   
</ul>
`;

exports[`Tabs Component only open one Accordion at a time 1`] = `
<ul
  className="Accordion"
>
  <li
    className="Accordion__item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
       
    </button>
  </li>
  <li
    className="Accordion__item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
       
    </button>
  </li>
  <li
    className="Accordion__item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
       
    </button>
    <p>
       
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
       
    </p>
  </li>
   
</ul>
`;

exports[`Tabs Component renders the first tab by default 1`] = `
<ul
  className="Accordion"
>
  <li
    className="Accordion__item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
       
    </button>
    <p>
       
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
       
    </p>
  </li>
  <li
    className="Accordion__item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
       
    </button>
  </li>
  <li
    className="Accordion__item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
       
    </button>
  </li>
   
</ul>
`;
